### PR TITLE
add note for CRD and RBAC handling for VPA (>=1.0.0)

### LINKED ISSUE #5921: Add instructions for updating VPA CRD for next release
#5911 (see also #5675) changes definition of VPA CRD by adding subresource.

I think this means update to the next VPA minor release will be more involved than usually.

I think usually it's ok to just change VPA deployments to use new images. This time we also need to change resource definition. We need to update resource definition in a way that doesn't break any existing VPAs (we don't want to delete existing VPA objects, users hopefully have important data there!).

I think we need clear instructions for updating CRD. Would be great to have instructions to roll back the change in case some need to go back to an older VPA versions. If those instructions are more than 1 command it would be nice to script them.

@wu0407 @kwiesmueller can you take a look?

### PR DESCRIPTION
**Pull request by wu0407**
#### What type of PR is this?
/kind documentation

#### What this PR does / why we need it:
This pull request addresses the need to update the Custom Resource Definition (CRD) and Role-Based Access Control (RBAC) configuration to support Vertical Pod Autoscaler (VPA) versions 0.15.0 and above. Additionally, it provides guidance on rolling back to version (<=0.14.0) if necessary.

#### Which issue(s) this PR fixes:

Fixes #5921 

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: vertical-pod-autoscaler/README.md [hunk 1]
```diff
@@ -65,6 +65,26 @@ The current default version is Vertical Pod Autoscaler 0.14.0
 | 0.4 to 0.7      | 1.11+              |
 | 0.3.X and lower | 1.7+               |
 
+### Notice on CRD update (>=0.15.0)
+**NOTE:** In version 0.15.0, we have updated the CRD definition and added RBAC for the status resource. If you are upgrading from version (<=0.14.0), you must update the CRD definition and RBAC.
```
Comment (Shubham82) [inline vertical-pod-autoscaler/README.md:69]:
```suggestion
**NOTE:** In version 0.15.0, we have updated the CRD definition and added RBAC for the
status resource. If you are upgrading from version (<=0.14.0), you must update the CRD
definition and RBAC.
```

File: vertical-pod-autoscaler/README.md [hunk 2]
```diff
@@ -65,6 +65,26 @@ The current default version is Vertical Pod Autoscaler 0.14.0
...
+```
+Another method is to re-execute the ./hack/vpa-up.sh script.
+```shell
+git clone https://github.com/kubernetes/autoscaler.git
+cd autoscaler/vertical-pod-autoscaler
+git checkout origin/vpa-release-0.15
+./hack/vpa-up.sh
+```
+
+If you need to rollback to version (<=0.14.0), please checkout the release for your rollback version and execute ./hack/vpa-up.sh. For example, to rollback to 0.14.0:
```
Comment (Shubham82) [inline vertical-pod-autoscaler/README.md:82]:
```suggestion
If you need to rollback to version (<=0.14.0), please checkout the release for your
rollback version and execute ./hack/vpa-up.sh. For example, to rollback to 0.14.0:
```

**Review by Shubham82 — CHANGES_REQUESTED** [review]

File: vertical-pod-autoscaler/README.md (hunk 1 above)
Comment (Shubham82) [reply vertical-pod-autoscaler/README.md:69]: @wu0407 it looks good to me. 
Just small nit: Wrap the line

File: vertical-pod-autoscaler/README.md (hunk 2 above)
Comment (Shubham82) [reply vertical-pod-autoscaler/README.md:82]: nit: Wrap the line

**Review by jbartosik — COMMENTED** [review]
Thank you. Did you test the update procedure(s)?

**Comment by sophieliu15** [comment]
> > Thank you. Did you test the update procedure(s)?
> 
> Sorry, I tested now and Modified the correct execution command. But found the duplicate install `vpa-recommender` in `recommender-deployment` and `recommender-externalmetrics-deployment`. I think needs to be fixed first.
> 
> https://github.com/kubernetes/autoscaler/blob/8e984d7c1a87740f43aa03b3cac00b2247c9e37c/vertical-pod-autoscaler/hack/vpa-process-yamls.sh#L43
> 
> 
> Lally Singh

Hope to follow up on this comment. Was the duplication fixed? If not, could you create an Issue for fixing this and add a comment in this PR to indicate it is blocked by the issue? Thank you!

**Comment by wu0407** [comment]
> > > Thank you. Did you test the update procedure(s)?
> > 
> > 
> > Sorry, I tested now and Modified the correct execution command. But found the duplicate install `vpa-recommender` in `recommender-deployment` and `recommender-externalmetrics-deployment`. I think needs to be fixed first.
> > https://github.com/kubernetes/autoscaler/blob/8e984d7c1a87740f43aa03b3cac00b2247c9e37c/vertical-pod-autoscaler/hack/vpa-process-yamls.sh#L43
> > 
> > Lally Singh
> 
> Hope to follow up on this comment. Was the duplication fixed? If not, could you create an Issue for fixing this and add a comment in this PR to indicate it is blocked by the issue? Thank you!

That is fixed by  https://github.com/kubernetes/autoscaler/pull/6151.
The current release version is 1.0.0. so need to update the version in this PR.

**Pull request merged**